src/tools/rbd_mirror/ImageSyncThrottler.h

```diff
@@ -33,56 +31,49 @@ namespace librbd { namespace journal { struct MirrorPeerClientMeta; } }
 namespace rbd {
 namespace mirror {
 
-template <typename> class ImageSync;
+template <typename> class InstanceWatcher;
 
 class ProgressContext;
 
 /**
  * Manage concurrent image-syncs
  */
 template <typename ImageCtxT = librbd::ImageCtx>
-class ImageSyncThrottler : public md_config_obs_t {
+class ImageSyncThrottler {
```

Comment:
Nit: should this class be the ```InstanceSyncThrottler``` since it's associated w/ the instance and the original version (now named ```InstanceSyncThrottler```) should remain the ```ImageSyncThrottler``` (or ```LeaderSyncThrottler```)? Right now, the ```InstanceXYZ``` classes are tied to a specific instance of rbd-mirror.

Alternatively, if you pass the ```InstanceWatcher``` reference down to the bootstrap state machine, it could instantly create a ```ImageSync``` state machine whole first state is just to invoke ```InstanceWatcher::notify_sync_start``` and wait. That would eliminate this pass-through class entirely.